scripting/pvpoptin.sp

```diff
@@ -622,23 +638,30 @@ public Action Command_ForcePvP(int client, int args) {
 	if (GetCmdArgs()!=2) {
 		char name[16];
 		GetCmdArg(0, name, sizeof(name));
-		ReplyToCommand(client, "Usage: %s <target|'map'> <1/0>", name);
+		ReplyToCommand(client, "Usage: %s <target|'map'> <-1/0/1/2>", name);
 	} else {
 		char pattern[MAX_NAME_LENGTH+1], tname[MAX_NAME_LENGTH+1];
 		GetCmdArg(2,pattern, sizeof(pattern));
-		bool pvpon = StringToInt(pattern) != 0;
+		bool pvplock = StringToInt(pattern) == 1;
+		bool pvpon = StringToInt(pattern) == 2;
+		bool pvpoff = StringToInt(pattern) == -1;
 		GetCmdArg(1,pattern, sizeof(pattern));
 		if (StrEqual(pattern, "map", false)) {
-			if (pvpon) globalPvP[0] |= State_Forced;
+			if (pvplock) globalPvP[0] |= State_Forced;
 			else globalPvP[0] &=~ State_Forced;
-			for (int i=1;i<=MaxClients;i++) {
+			// I have no idea why you would want to turn off someone's forced state if you just turn off map forced pvp.

```

Comment:
'map' is for late joiners

Reply:
!forcepvp map is not the same as !forcepvp @all though. I think of it as being similar as using a convar to turn on crits for the entire server (i.e. allcrits).